packages/protocol/contracts/stability/PairManager.sol

```diff
@@ -0,0 +1,290 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { IPairManager } from "./interfaces/IPairManager.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+import { IMentoExchange } from "./interfaces/IMentoExchange.sol";
+
+import { StableToken } from "./StableToken.sol";
+
+import { UsingRegistry } from "../common/UsingRegistry.sol";
+import { Initializable } from "../common/Initializable.sol";
+import { FixidityLib } from "../common/FixidityLib.sol";
+
+// TODO: Remove when migrating to mento-core. Newer versions of OZ-contracts have this interface
+interface IERC20Metadata {
+  function symbol() external view returns (string memory);
+}
+
+/**
+ * @title PairManager
+ * @notice The pair manager allows the creation of virtual pairs that are tradeable via Mento.
+ */
+contract PairManager is IPairManager, Initializable, UsingRegistry {
+  using FixidityLib for FixidityLib.Fraction;
+
+  /* ==================== State Variables ==================== */
+
+  // Address of the broker contract.
+  address public broker;
+
+  // Maps a pair id to the corresponding pair struct.
+  // pairId is in the format "stableSymbol:collateralSymbol:exchangeName"
+  mapping(bytes32 => Pair) public pairs;
+  // Tracks whether or not a pair with the given pairId exists.
+  mapping(bytes32 => bool) public isPair;
+
+  // TODO: Implement getPairs that takes in asset addresses & not exchange
+
+  // Tracks the total fraction of reserve collateral
+  // that has been allocated to existing virtual pairs.
+  mapping(address => FixidityLib.Fraction) public allocatedCollateralFractions;
+
+  // TODO: https://github.com/mento-protocol/mento-general/issues/60
+  // Tracks the total fraction of reserve stable that has been allocated to existing virtual pairs.
+  mapping(address => FixidityLib.Fraction) public allocatedStableFractions;
```

Comment:
Honestly I'd remove all this code around collateral fraction / stable fraction, as far as I can see both these variables and usage of the related functions is commented out, so just to make this clean I'd remove all of this code and take care of everything in a later issue.